---
title: "Lab 2 - UCP1 Variants and Hardy-Weinberg Equilibrium"
author: "Becca DeCamp"
date: "6/4/2018"
output: html_document
---
## Introduction to Hardy-Weinberg Equilibrium Lab
<br>

We have already discussed Hardy-Weinberg Equilibrium in class, but we haven't really linked it to population genetics yet. Today's lab will focus on the uses of Hardy-Weinberg Equilibrium in modern population genetics as we use R to look at Hardy-Weinberg Equilibrium in our assigned populations. 
<br>

### Hardy-Weinberg Equilibrium in Population Genetics
<br>
In modern population genetics studies, Hardy-Weinberg Equilibrium has a slightly different meaning than what we have talked about in class. A population that is in true Hardy-Weinberg Equilibrium has to fit five conditions: The population must be extremely large, it must be isolated from other populations, there must be no mutations, there must be random mating, and there must be no natural selection. Obviously, holding a population of any species to this standard is impossible, so we will use Hardy-Weinberg differently in our population genetics study. 
<br>
<br>
You will often see Hardy-Weinberg Equilibrium being used in modern population genetics to simply measure whether the observed genotype frequencies of a specific SNP or microsatellite are the same as the estimated genotype frequencies. Population geneticists will often use this measure of deviance as a "starting point," so to speak, for their studies. They use Hardy-Weinberg as their starting point for a few reasons; to check the genotype quality of their data, to get a general sense of the amount of allelic variability and selection happening at specific loci, or, most commonly, as an assumption about the frequency of disease causing genotypes. The assigned Rose et al. paper use Hardy- Weinberg in the latter manner as a check to make sure there was no selection happening at the two loci they looked at.
<br>
<br>
In order to determine how much the observed phenotype deviates from the espected phenotype, a statistical test called a Chi Square Goodness Of Fit test is usually performed, the output of which is a P-value that will tell you how much the observed genotype frequency deviates from the expected frequency. However, we learned from the Wigginton reading that the Chi Square test has a tendency to give false positives. In their paper, they proposed a new mathematical method of calculating the "True" Hardy-Weinberg value, which fixes the problem of false positives. We will not need to get in to the specifics of the equation to calculate the "True" Hardy-Weinberg Equilibrium, but you will be calculating your population's "True" Hardy-Weinberg Equilibrium as well as using a traditional Chi Square test. 
<br>
<br>
Now that we have learned more about the applications of Hardy-Weinberg Equilibrium in modern population genetics, let's get to the lab! 
<br>

### Learning Objectives
<br>

* Use the SCC and R coding language to observe and understand population differences in UCP1 variation.

* Calculate Hardy-Weinberg Equilibrium for all UCP1 SNPs in individual populations using a traditional Chi Square test.

* Perform a check on all SNPs not in Hardy-Weinberg Equilibrium by calculating "True" Hardy-Weinberg with the built-in Shiny App. 

* Research the consequence types of these SNPs in order to understand how these SNPs might affect the genome itself, and how they might affect genotype. 

* Calculate the "True" Hardy-Weinberg Equilibrium using the Shiny App for A-3826G (SNP ID rs1800592), an upstream SNP that has known phenotypic consequences, and determine what the Hardy-Weinberg Equilibrium says about the population. 